corehq/apps/hqadmin/service_checks.py

```diff
@@ -0,0 +1,170 @@
+        mq_management_url = amqp_parts[0].replace('5672', '15672')
+        vhost = amqp_parts[1]
+        try:
+            mq = Resource('http://%s' % mq_management_url, timeout=2)
+            vhost_dict = json.loads(mq.get('api/vhosts', timeout=2).body_string())
+            for d in vhost_dict:
+                if d['name'] == vhost:
+                    return ServiceStatus(True, 'RabbitMQ OK')
+            return ServiceStatus(False, 'RabbitMQ Offline')
+        except Exception as e:
+            return ServiceStatus(False, "RabbitMQ Error: %s" % e)
+    else:
+        return ServiceStatus(False, "RabbitMQ Not configured")
+
+
+def check_pillowtop():
+    return ServiceStatus(False, "Not implemented")
+
+
+@change_log_level('kafka.client', logging.WARNING)
+def check_kafka():
+    client = get_kafka_client_or_none()
+    if not client:
+        return ServiceStatus(False, "Could not connect to Kafka")
+    return ServiceStatus(True, "Kafka's fine. Probably.")
+
+
+def check_touchforms():
+    try:
+        res = requests.post(settings.XFORMS_PLAYER_URL,
+                            data='{"action": "heartbeat"}',
+                            timeout=5)
+    except requests.exceptions.ConnectTimeout:
+        return ServiceStatus(False, "Could not establish a connection in time")
+    except requests.ConnectionError:
+        return ServiceStatus(False, "Could not connect to touchforms")
+    else:
+        msg = "Touchforms returned a {} status code".format(res.status_code)
+        return ServiceStatus(res.ok, msg)
+
+
+@change_log_level('urllib3.connectionpool', logging.WARNING)
+def check_elasticsearch():
+    doc = {'_id': 'elasticsearch-service-check',
+           'date': datetime.datetime.now().isoformat()}
+    send_to_elasticsearch('groups', doc)
+    time.sleep(1)
+    hits = GroupES().remove_default_filters().doc_id(doc['_id']).run().hits
+    send_to_elasticsearch('groups', doc, delete=True)  # clean up
+    if doc in hits:
+        return ServiceStatus(True, "Successfully sent a doc to ES and read it back")
+    return ServiceStatus(False, "Something went wrong sending a doc to ES")
+
+
+def check_shared_dir():
+    return ServiceStatus(False, "Not implemented")
+
+
+def check_blobdb():
+    """Save something to the blobdb and try reading it back."""
+    db = get_blob_db()
+    contents = "It takes Pluto 248 Earth years to complete one orbit!"
+    info = db.put(StringIO(contents))
+    with db.get(info.identifier) as fh:
+        res = fh.read()
+    db.delete(info.identifier)
+    if res == contents:
+        return ServiceStatus(True, "Successfully saved a file to the blobdb")
+    return ServiceStatus(False, "Failed to save a file to the blobdb")
+
+
+def check_celery():
+    from celery import Celery
+    from django.conf import settings
+    celery = Celery()
+    celery.config_from_object(settings)
+    worker_responses = celery.control.ping(timeout=10)
```

Comment:
This will wait a full 10 seconds (or maybe until all workers have responded). I think for this check it would be sufficient to report success after a single response, which could be achieved by passing `limit=1`. Might want to update the success message below if you do that.